mdata/store_cassandra.go

```diff
@@ -73,13 +76,68 @@ It's safe for concurrent use by multiple goroutines and a typical usage scenario
 object to interact with the whole Cassandra cluster.
 */
 
+type ttlTables map[uint32]ttlTable
+type ttlTable struct {
+	table  string
+	window uint32
+}
+
 type cassandraStore struct {
 	session     *gocql.Session
 	writeQueues []chan *ChunkWriteRequest
 	readQueue   chan *ChunkReadRequest
+	ttlTables   ttlTables
+}
+
+func ttlUnits(ttl uint32) float64 {
+	// convert ttl to hours
+	return float64(ttl) / (60 * 60)
+}
+
+func getTTLTables(ttls []uint32, windowFactor int, nameFormat string) ttlTables {
+	tables := make(ttlTables)
+	for _, ttl := range ttls {
+		/*
+		 * the purpose of this loop is to bucket metrics of similar TTLs.
+		 * we first calculate the largest power of 2 that's smaller than the TTL and then divide the result by
+		 * the window factor. for example with a window factor of 20 we want to group the metrics like this:
+		 *
+		 * generated with: https://gist.github.com/replay/69ad7cfd523edfa552cd12851fa74c58
+		 *
+		 * +-----------------------------+---------------+-------------+----------+
+		 * |                hours / days |    table_name | window_size | sstables |
+		 * +-----------------------------+---------------+-------------+----------+
+		 * |                      1 / <1 |     metrics_1 |           1 |    1 - 1 |
```

Comment:
if the time span can be `39.9847`, and the `ts` of a datapoint is in the middle of the table, then the end of live of that datapoint will be in the middle of the `41st` table. So that comes from the fact that the `ts` don't need to be aligned with the table wrapping times.